test_conformance/api/test_kernel_local_memory_size.cpp

```diff
@@ -0,0 +1,259 @@
+        return TEST_FAIL;
+    }
+
+    kernel_local_usage = 0;
+    error = clGetKernelWorkGroupInfo(kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE,
+                                     sizeof(kernel_local_usage),
+                                     &kernel_local_usage, NULL);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    const size_t size = 10;
+    const size_t memory = size * sizeof(cl_int);
+
+    size_t global_work_size[] = { size };
+
+    std::unique_ptr<int[]> data(new int[size]);
+    for (size_t i = 0; i < size; i++)
+    {
+        data[i] = 0;
+    }
+    clMemWrapper streams[2];
+
+    streams[0] =
+        clCreateBuffer(context, CL_MEM_READ_WRITE, memory, NULL, &error);
+    test_error(error, "Creating test array failed");
+
+    error = clSetKernelArg(kernel, 0, sizeof(streams[0]), &streams[0]);
+    test_error(error, "Unable to set indexed kernel arguments");
+
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, nullptr, global_work_size,
+                                   nullptr, 0, NULL, nullptr);
+    test_error(error, "clEnqueueNDRangeKernel failed.");
+
+    error = clEnqueueReadBuffer(queue, streams[0], CL_TRUE, 0, memory,
+                                data.get(), 0, NULL, NULL);
+    test_error(error, "clEnqueueReadBuffer failed");
+
+    test_assert_error(kernel_local_usage >= memory,
+                      "kernel local mem size failed");
+
+    // Check memory needed to execute empty kernel with __local parameter with
+    // setKernelArg
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    &local_param_kernel, "local_param_kernel")
+        != 0)
+    {
+        return TEST_FAIL;
+    }
+
+    kernel_local_usage = 0;
+
+    for (size_t i = 0; i < size; i++)
+    {
+        data[i] = i;
+    }
+
+    streams[0] = clCreateBuffer(context, CL_MEM_COPY_HOST_PTR, memory,
+                                data.get(), &error);
+    test_error(error, "Creating test array failed");
+    streams[1] =
+        clCreateBuffer(context, CL_MEM_READ_WRITE, memory, nullptr, &error);
+    test_error(error, "Creating test array failed");
+
+    error = clSetKernelArg(kernel, 0, memory, NULL);
+    test_error(error, "Unable to set indexed kernel arguments");
+    error = clSetKernelArg(kernel, 1, sizeof(streams[0]), &streams[0]);
+    test_error(error, "Unable to set indexed kernel arguments");
+    error = clSetKernelArg(kernel, 2, sizeof(streams[1]), &streams[1]);
+    test_error(error, "Unable to set indexed kernel arguments");
+
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, nullptr, global_work_size,
+                                   nullptr, 0, NULL, nullptr);
+    test_error(error, "clEnqueueNDRangeKernel failed.");
+
+    error = clEnqueueReadBuffer(queue, streams[1], CL_TRUE, 0, memory,
+                                data.get(), 0, NULL, NULL);
+    test_error(error, "clEnqueueReadBuffer failed");
```

Comment:
We're not validate the result that gets written back to `data`. I'm not sure we actually need to be executing kernels at all to be verifying the `CL_KERNEL_LOCAL_MEM_SIZE ` query. But if we are doing it for a more thorough test, then we may as well verify that `data` contains the values we expect.

Reply:
@EwanC added verification